### ISSUE TITLE
Crutches.Format.Number.as_percentage

### ISSUE THREAD

**Issue by mykewould** [mykewould/crutches#27]
```
@doc ~S"""
Formats a `number` as a percentage string (e.g., 65%). You can customize the 
format in the `options` Dict

The `as_percentage!` method raises an error if the input is not numeric.

# Options

 * `:locale` - Sets the locale to be used for formatting (defaults to current locale).
 * `:precision` - Sets the precision of the number (defaults to 3).
 * `:significant` - If true, precision will be the # of significant_digits. If false, the # of fractional digits (defaults to false).
 * `:separator` - Sets the separator between the fractional and integer digits (defaults to “.”).
 * `:delimiter` - Sets the thousands delimiter (defaults to “”).
 * `:strip_insignificant_zeros` - If true removes insignificant zeros after the decimal separator (defaults to false).
 * `:format` - Specifies the format of the percentage string The number field is %n (defaults to “%n%”).

# Examples

    iex> Number.as_percentage(100)                                        
    "100.000%"

    iex> Number.as_percentage("98")                                       
    "98.000%"

    iex> Number.as_percentage(100, precision: 0)                          
    "100%"

    iex> Number.as_percentage(1000, delimiter: '.', separator: ',')       
    "1.000,000%"

    iex> Number.as_percentage(302.24398923423, precision: 5)              
    "302.24399%"

    iex> Number.as_percentage(1000, locale: :fr)                          
    "1 000,000%"

    iex> Number.as_percentage("98a")                                      
    "98a%"

    iex> Number.as_percentage(100, format: "%n  %")                       
    "100  %"

    iex> Number.as_percentage!("98a", raise: true)                        
    ** (ArithmeticError) bad input number

"""
```

**Comment by Axxiss** [comment]
Hi, just to avoid stepping on each others work. I intend to implement this issue.

As i18n is still discussed in #40, locale parameter will have no effect until a something clearer came up.

**Comment by druzn3k** [comment]
@Axxiss it's yours. We learned about the existence of [Number](https://github.com/danielberkompas/number) and already talked with @danielberkompas about a possible convergence of his library with ours. For now we are going our direction without a merge in the _imminent_ future (not before 1.0 at least) because we still are a learning project, and merging will take away a lot of learning material. The only thing asked is, if you take anything from his repo, that his name should be included in the README. I advise against diving in his code too early, as you will be biased on how to build this function, but if you are stuck somehow in the near future, know that his library is really good and well documented.

TL;DR: @Axxiss it's yours, you could take inspiration from [Number](https://github.com/danielberkompas/number), but in the spirit of learning please avoid coping verbatim. :+1:

**Comment by Axxiss** [comment]
Great, will try to came up with a first version on my own. Then we can compare/merge/refactor against Number version.

**Comment by druzn3k** [comment]
Good luck @Axxiss :+1:

**Comment by Axxiss** [comment]
Given this case

```
iex> Number.as_percentage(100, format: "%n  %")                       
"100  %"
```

The expected output shouldn't be `100.000 %` as the default `precision` is 3, and `strip_insignificant_zeros` is false.

**Comment by druzn3k** [comment]
The choice is yours, the only _rule_ is that it has to be consistent with the other parts of the library. If `%n` has been used to reflect the standard options in the other methods, go with it. :+1:

**Comment by Axxiss** [comment]
I think is more than reflect the behaviour of other format method, all methods should behave the same. It doesn't make sense that `as_percentage(100) == 100.000%` and  `as_percentage(100, format: '%n %') == "100 %"`. In that case for the same numerical input we have two different representations (leaving aside the % symbol).

As a user I would expect that a method does the same, despite the params I pass to it.

**Comment by duijf** [comment]
This was merged with #59.
